OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -151,7 +151,7 @@ public void AddSmudge(CPos loc)
 			else
 			{
 				// Existing smudge; make it deeper
-				var tile = dirty.ContainsKey(loc) ? dirty[loc] : tiles[loc];
```

Comment:
Can you please add a comment '// .Sprite will be null if the smude has just been removed'?